systran_storages/storages/corpus.py

```diff
@@ -409,24 +409,35 @@ def push_file(self, local_path, remote_path):
                 raise ValueError(
                     'Cannot push %s, only support format of the corpus (application/x-tmx+xml, '
                     'text/bitext)' % local_path)
+            source = lp.get('source', '')
+            targets = lp.get('targets', [])
+            import_options = {
+                "cleanFormatting": True,
+                "removeDuplicates": True,
+                "expectedSourceLanguage": source,
+                "expectedTargetLanguages": targets
+            }
 
             mp_encoder = MultipartEncoder(
                 [
                     ('accountId', self.account_id),
                     ('format', format_path),
-                    ('importOptions', '{"cleanFormatting": true, "removeDuplicates":true}'),
+                    ('importOptions', json.dumps(import_options)),
                     ('filename', self._create_path_from_root(remote_path)),
                     ('corpus', data)
                 ]
             )
             response = requests.post(self.host_url + '/corpus/import', data=mp_encoder,
                                      headers={'Content-Type': mp_encoder.content_type})
             if response.status_code != 200:
+                error_message = json.loads(response.content).get('error')
```

Comment:
In case of error, for example: source/targets is not expected, what is the current behavior of CM2?
(The error corpus still is imported ? or it's deleted later?)